### PR TITLE
Fedora build

### PR DESCRIPTION
**Pull request by eidins**
Replacement for https://github.com/open-eid/libdigidocpp/pull/410 with rebase and conflict resolved

### CODE REVIEW

**Comment by eidins** [comment]
```
cmake ..
make
make package
```

All work and the RPM installs fine.

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,16 @@ set( BUILD_TOOLS YES CACHE BOOL "Build digidoc-tool" )
...
 set( SIGNCERT "" CACHE STRING "Common name of certificate to used sign binaries, empty skip signing" )
 set( CROSSSIGNCERT "" CACHE STRING "Common name of certificate to used cross sign binaries, empty skip signing" )
+set(CPACK_PACKAGE_VERSION ${VERSION})
+set(CPACK_GENERATOR "RPM" "DEB")
+set(CPACK_PACKAGE_NAME "libdigidocpp")
+set(CPACK_PACKAGE_RELEASE 1)
+set(CPACK_PACKAGE_CONTACT "Andrus Randveer")
+set(CPACK_PACKAGE_VENDOR "RIA")
+set(CPACK_PACKAGING_INSTALL_PREFIX "/usr/")
+set(CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/usr/local" "/usr/local/etc" "/usr/share/man" "/usr/share/man/man1" "/usr/lib64/pkgconfig")
```
Comment (metsma) [inline CMakeLists.txt:39]: Package should not install anything into /usr/local

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -60,6 +60,40 @@ jobs:
       with:
         name: pkgs
         path: macOS/libdigidocpp*.*
+  fedora:
+    name: Build on Fedora to ${{ matrix.container }}
+    runs-on: fedora-latest
```
Comment (metsma) [inline .github/workflows/build.yml:65]: ubuntu-latest, Github action does not have fedora-latest

**Review by metsma — CHANGES_REQUESTED** [review]

**Comment by metsma** [comment]
If you plan build digidoc4-client you probably need also save the artifacts. If not we can merge

**Review by metsma — APPROVED** [review]

**Comment by metsma** [comment]
There are some issues with the artefacts
"Build on Fedora to fedora:33
No files were found with the provided path: libdigidocpp*.rpm. No artifacts will be uploaded."

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -60,6 +60,40 @@ jobs:
...
+        mkdir build
+        cd build
+        cmake ..
+        make
+        make package
+    - name: Archive artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: rpms
+        path: libdigidocpp*.rpm
```
Comment (metsma) [inline .github/workflows/build.yml:96]: Build on Fedora to fedora:33
No files were found with the provided path: libdigidocpp*.rpm. No artifacts will be uploaded.

Comment (metsma) [reply .github/workflows/build.yml:96]: path: build/libdigidocpp*.rpm ??

**Comment by metsma** [comment]
Lets see if this also adds config files to correct place
set(CPACK_RPM_SOURCE_PKG_BUILD_PARAMS -DCMAKE_INSTALL_SYSCONFDIR=/etc)

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -60,6 +60,40 @@ jobs:
...
+      run: |
+        dnf install -y \
+          cmake openssl-devel xerces-c-devel \
+          xml-security-c-devel zlib-devel vim-common \
+          https://www.codesynthesis.com/download/xsd/4.0/linux-gnu/x86_64/xsd-4.0.0-1.x86_64.rpm
+    - name: Build
+      run: |
+        mkdir build
+        cd build
+        cmake ..
```
Comment (metsma) [inline .github/workflows/build.yml:89]:
```suggestion
        cmake -DCMAKE_INSTALL_SYSCONFDIR=/etc ..
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -29,6 +29,16 @@ set( BUILD_TOOLS YES CACHE BOOL "Build digidoc-tool" )
...
 set( SIGNCERT "" CACHE STRING "Common name of certificate to used sign binaries, empty skip signing" )
 set( CROSSSIGNCERT "" CACHE STRING "Common name of certificate to used cross sign binaries, empty skip signing" )
+set(CPACK_PACKAGE_VERSION ${VERSION})
+set(CPACK_GENERATOR "RPM" "DEB")
+set(CPACK_PACKAGE_NAME "libdigidocpp")
+set(CPACK_PACKAGE_RELEASE 1)
+set(CPACK_PACKAGE_CONTACT "Andrus Randveer")
+set(CPACK_PACKAGE_VENDOR "RIA")
+set(CPACK_PACKAGING_INSTALL_PREFIX "/usr/")
+set(CPACK_RPM_SOURCE_PKG_BUILD_PARAMS -DCMAKE_INSTALL_SYSCONFDIR=/etc)
```
Comment (metsma) [inline CMakeLists.txt:39]:
```suggestion
```
Seems like this did not help

**Review by metsma — CHANGES_REQUESTED** [review]

**Review by metsma — APPROVED** [review]

**Pull request merged**